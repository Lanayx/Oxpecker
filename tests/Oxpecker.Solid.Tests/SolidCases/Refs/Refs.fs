module Oxpecker.Solid.Tests.Cases.Refs

open Browser.Types
open Oxpecker.Solid
open Fable.Core.JS


[<SolidComponent>]
let Test () =
    let mutable htmlCanvas: HTMLCanvasElement = Unchecked.defaultof<_>
    onMount(fun () -> console.log(htmlCanvas.height + htmlCanvas.width))

    div().ref(fun _ -> console.log("before mounted")) {
        canvas(width=256, height=256).ref(htmlCanvas)
    }
