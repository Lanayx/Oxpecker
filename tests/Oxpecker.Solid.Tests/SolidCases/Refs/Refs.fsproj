<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="Refs.fs" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\..\..\src\Oxpecker.Solid\Oxpecker.Solid.fsproj" />
  </ItemGroup>

</Project>
