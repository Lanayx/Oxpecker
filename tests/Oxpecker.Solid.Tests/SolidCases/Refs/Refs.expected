import { defaultOf } from "./fable_modules/fable-library-js.4.23.0/Util.js";
import { onMount } from "solid-js";
import { some } from "./fable_modules/fable-library-js.4.23.0/Option.js";

export function Test() {
    let htmlCanvas = defaultOf();
    onMount(() => {
        console.log(some(htmlCanvas.height + htmlCanvas.width));
    });
    return <div ref={(_arg) => {
            console.log(some("before mounted"));
        }}>
        <canvas width={256}
            height={256}
            ref={htmlCanvas}></canvas>
    </div>;
}

